storage/range.go

```diff
@@ -614,7 +614,10 @@ func (r *Range) addReadOnlyCmd(ctx context.Context, args proto.Request, reply pr
 		if header.Timestamp.Equal(proto.ZeroTimestamp) {
 			header.Timestamp = r.rm.Clock().Now()
 		}
-		intents, err := r.executeCmd(r.rm.Engine(), nil, args, reply)
+		executedReply, intents, err := r.executeCmd(r.rm.Engine(), nil, args)
+		executedReply.Header().SetGoError(err)
```

Comment:
Before calling `SetGoError` we should assert that `executedReply.Header().GoError()` is nil.